spring-content-s3/src/test/java/internal/org/springframework/content/s3/it/S3StoreWithEntityConverterIT.java

```diff
@@ -0,0 +1,397 @@
+    }
+
+    @Configuration
+    @EnableJpaRepositories(basePackages="internal.org.springframework.content.s3.it", considerNestedRepositories = true)
+    @EnableS3Stores(basePackages="internal.org.springframework.content.s3.it")
+    @Import(InfrastructureConfig.class)
+    public static class TestConfig {
+
+        @Bean
+        public S3Client client() throws URISyntaxException {
+            s3ClientSpy = spy(LocalStack.getAmazonS3Client());
+            return s3ClientSpy;
+        }
+    }
+
+    @Configuration
+    public static class CustomConverterConfig {
+
+      @Bean
+      public S3StoreConfigurer configurer() {
+          return new S3StoreConfigurer() {
+
+              @Override
+              public void configureS3StoreConverters(ConverterRegistry registry) {
+
+                  registry.addConverter(new Converter<ContentPropertyInfo<TestEntity, Serializable>, S3ObjectId>() {
+                      @Override
+                      public S3ObjectId convert(ContentPropertyInfo<TestEntity, Serializable> info) {
+                          return new S3ObjectId(OTHER_OTHER_BUCKET, info.contentId().toString());
+                      }
+                  });
+
+
+                  registry.addConverter(new Converter<ContentPropertyInfo<FakeEntity, Serializable>, S3ObjectId>() {
+                      @Override
+                      public S3ObjectId convert(ContentPropertyInfo<FakeEntity, Serializable> info) {
+                          throw new IllegalStateException("wrong converter called");
+                      }
+                  });
+              }
+          };
+       }
+    }
+
+    @Configuration
+    public static class InfrastructureConfig {
+
+        @Bean
+        public DataSource dataSource() {
+            EmbeddedDatabaseBuilder builder = new EmbeddedDatabaseBuilder();
+            return builder.setType(EmbeddedDatabaseType.H2).build();
+        }
+
+        @Bean
+        public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
+
+            HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
+            vendorAdapter.setDatabase(Database.H2);
+            vendorAdapter.setGenerateDdl(true);
+
+            LocalContainerEntityManagerFactoryBean factory = new LocalContainerEntityManagerFactoryBean();
+            factory.setJpaVendorAdapter(vendorAdapter);
+            factory.setPackagesToScan("internal.org.springframework.content.s3.it");
+            factory.setDataSource(dataSource());
+
+            return factory;
+        }
+
+        @Bean
+        public PlatformTransactionManager transactionManager() {
+
+            JpaTransactionManager txManager = new JpaTransactionManager();
+            txManager.setEntityManagerFactory(entityManagerFactory().getObject());
+            return txManager;
+        }
+    }
+
+    public static class FakeEntity {
+    }
+
+    @Entity
+    @Setter
+    @Getter
+    @NoArgsConstructor
+    @Table(name="test_entity")
```

Comment:
@paulcwarren Thank you for merging!

I see that some tests in `S3StoreIT` do not pass now, although I ran the tests locally before submitting of the PR (but maybe not this class, as I thought it was unrelated to changes) and my PR passed checks too (apparently tests are not always part of PR checks).

The current error is related to this line, where the same table name `@Table(name="test_entity")` is used for different Entity types. And when you run `S3StoreIT` it actually generates DDL for the `test_entity` table corresponding to `S3StoreWithEntityConverterIT.TestEntity`, where the `UUID contentId` is mapped to `binary`. When you try to save `S3StoreIT.TestEntity` instance to this table, a `Hexadecimal string contains non-hex character` error appears because  `contentId` in `S3StoreIT.TestEntity` is of type `String`. It's probably easiest to remove this annotation everywhere or use unique table names for different Entities.

By the way, AzureStorageWithEntityConverterIT.java added another Entity with `@Table(name="test_entity")`, which can also cause similar problems.